### ISSUE TITLE
Deck options: 'back' should only request confirmation if changes are made

### ISSUE THREAD

**Issue by Arthur-Milchior** [ankidroid/Anki-Android#14438]
Running on HEAD. If you open the deck option and change them then press back they are discarded. Please instead ask whether they should be cancelled or saved.

(Also, the save button is quite small. I doubt it satisfies accessibility requirement)

**Comment by dae** [comment]
The first point would require some work to implement, as currently the frontend doesn't have a flag that tracks when a change has been made. You'd need to store a copy of the initial data, then provide some JS method that compares the current state with the initial state to see if anything has changed. The Kotlin code would then need to invoke it and decide what to do based on its response.

**Comment by SanjaySargam** [comment]
Working on it

**Comment by david-allison** [comment]
User comment:

> Every time we tap Save button, the Deck options screen closes and we need to go back again and again to Deck options to continue our job. I think the Save button is not necessary or at least it shouldn’t close the screen.

**Comment by briankrznarich** [comment]
I was actually coming here to consider filing a similar ticket.  The title would have been something like:
"Inconsistent back-button behavior between AnkiDroid core settings and Deck settings" .

For the internal app settings (hamburger menu->settings), "back" will unconditionally save, which I think is how Ankidroid has always worked, even for deck-specific settings.  Now, as you know, deck settings require the user to tap "save", or your changes will be silently discarded.

I screwed this up 4 or 5 times before I realized why my settings changes were not sticking.   It doesn't take much work to adapt to, but *every* ankidroid user is going to screw this up 4 or 5 times before they figure out what's going on.   A confirmation warning would alleviate this problem.

A long time ago I disliked this "back = unconditional save" mechanic, with no way to easily undo accidental changes.  But all of iOS works this way, and most of Android I think.  People are losing an intuition for an explicit "Save" button on phones, especially a tiny button at the top-right of the screen.

I agree with the earlier remarks on button-size and accessibility.   The "save" button, and *especially* the drop-down menu to the right of the "save" button, are very very small.  Besides being made physically taller, padding would probably help.  There is literally 0 padding between the right side of the "Save" button and the edge of the screen.  Whitespace on the right would make a safer tap-target for the drop-down menu.

**Comment by david-allison** [comment]
@briankrznarich Thank you for your time, I agree and I'll get this resolved

**Comment by david-allison** [comment]
Keep this open until we ask for confirmation on change, rather than unconditionally

**Comment by user1823** [comment]
Still looking for a solution...

**Comment by Arthur-Milchior** [comment]
I just opened this PR https://github.com/ankitects/anki/pull/3410 , if DAE accepts it, once it's included in AnkiDroid, we can query the javascript to know whether there was any change.

**Comment by user1823** [comment]
The backend changes have been incorporated into the AnkiDroid `main` branch. So, the interested devs can work on this now.

**Comment by david-allison** [comment]
@Arthur-Milchior 

Was this fixed in the following? 
* https://github.com/ankidroid/Anki-Android/pull/17315

**Comment by Arthur-Milchior** [comment]
Yes but it'll be broken with anki backend change because I removed the Bridge command that I introduced before knowing it was deprecated

**Comment by david-allison** [comment]
That's fine. We won't update the backend until they start on RCs

**Comment by user1823** [comment]
To clarify, Arthur didn't remove the implementation from upstream. He replaced it with another one. Here's the PR: https://github.com/ankitects/anki/pull/3571

So, I guess AnkiDroid would have to modify some of its own code after the backend update.

**Comment by david-allison** [comment]
Things move fast! This is now actively in the pipeline:

* https://github.com/ankidroid/Anki-Android-Backend/issues/456
